### ISSUE TITLE
UX/UI to show plugins not available for install from the bundled app

### ISSUE THREAD

**Issue by junxini** [napari/napari#4145]
### Overview
The bundle application will need to inform the user of plugins that are not available for installation. There’s currently a list of 142 plugins, not all of which will be available within the app (approximately 20 will not be available, tbc). The design is meant to show users all plugins that are available for install, and ones that are not yet available with a short explanation (can be standardized across all). 

### What level of design is needed? (Choose all that apply)
- 🔲   **User experience research:**  high-level recommendation/exploration of user needs, design heuristics, and / or best practices to inform a design experience
(Use this option when you feel there’s a challenge to be solved, but you’re curious about what the experience should be — may involve research studies to understand challenges/opportunities for design)
- 🔲   **information flow / conceptual:** organizing and structuring of information flow and content, including layout on screen or across multiple steps
- ✅  **Visual:** creating mockups, icons, etc
(If choosing this level alone, it means that the content to be mocked up and its organization is already known and specified)

### Is design a blocker?
- 🔲  **Yes:** engineering cannot proceed without a design first
- ✅   **No:** engineering can create a first version, and design can come in later to iterate and refine

**Comment by tlambert03** [comment]
not totally following here.  _no_ plugins come pre-installed with the bundled app.  And for those that are "uninstallable" can't we just not show them in the available list?  (or, if you're trying to avoid someone from saying "why isn't this plugin showing up??" maybe show them grayed out with an icon on them?)

**Comment by goanpeca** [comment]
>no plugins come pre-installed with the bundled app.

Well, the console does, which does appear as a plugin, no?

>And for those that are "uninstallable" can't we just not show them in the available list? (or, if you're trying to avoid someone from saying "why isn't this plugin showing up??" maybe show them grayed out with an icon on them?)

That is what we are trying to figure out, we could just not show them but maybe showing them on a greyed out fashion is better for communicating to users that it will become available eventually?

**Comment by isabela-pf** [comment]
@goanpeca gave me a great segue!

As I think I’ve heard, there’s a few layers of potential feedback for this.
- Labeling which plugins can’t be installed
- Giving general feedback on why plugins might not be able to be installed
- Giving specific feedback on why individual plugins can’t be installed

And I’m not sure the workload required for each level. I personally think labeling and providing a link to general reasons why plugin might not be installable seems like the best amount for a first pass, though I’d be happy to know if we should be accounting for more specificity.

With this in mind, I think the best solutions will use the warning icon already used in napari like the icon labeling npe2 plugins:

<img width="918" alt="napari plugin dialog with one item adding a warning icon and a link labelled cannot install plugin." src="https://user-images.githubusercontent.com/50221806/155219255-b2b19fe4-89b7-4d5e-b84d-3f2cb9900c72.png">

I also think it’d be ideal to have the labeling text link out to documentation for troubleshooting why a plugin might not be installable. I don’t think that documentation exists yet, though? In that case only the icon with a tool tip may be best.

https://user-images.githubusercontent.com/50221806/155219845-0ee09364-d1fe-458b-a4f8-9bbf4719f62c.mov

(Video description: An item on the napari plugin dialog with a warning icon and the tool tip `Cannot install plugin`)

To summarize, plugins that can’t be installed will look like other plugins listed except for the following styling:
- Add standard napari warning icon (with tool tip)
- Install button is disabled
- If possible, link out to troubleshooting plugins that can't be installed.

**Comment by tlambert03** [comment]
sorry, yes of course you're correct @goanpeca  😊 
@isabela-pf ... i love the icon and general idea
@goanpeca: do you know whether the API is going to have this info (such as a set of requirements that must be met)?  or is this incompatibility something we'll need to determine at runtime (in napari based on the environment)?

**Comment by psobolewskiPhD** [comment]
I know this is about the bundle, but would be super cool if napari could help ID plugins with compiled/heavy depends and flag them with something to tell the user that this will need more work. Frequently those plugins have partner repos with detailed install instructions...
And in the new conda bundle, the magics %pip and %conda can actually go a long way to solving some of those issues...

**Comment by goanpeca** [comment]
>do you know whether the API is going to have this info (such as a set of requirements that must be met)? or is this incompatibility something we'll need to determine at runtime (in napari based on the environment)?

I think eventually, yes that is the idea we have discussed with @neuromusic. We can have a lot of info (compatibility with a given napari version, availability for a given operating system) but on the first pass I would say we will not have it in place for the upcoming release.

**Comment by neuromusic** [comment]
☝️  what @goanpeca said

**Comment by DragaDoncila** [comment]
>I know this is about the bundle, but would be super cool if napari could help ID plugins with compiled/heavy depends and flag them with something to tell the user that this will need more work. Frequently those plugins have partner repos with detailed install instructions...

This is a great point @psobolewskiPhD. We've discussed similar ideas in the context of displaying metadata on the hub e.g. whether plugin is installable by napari, whether its tests run, etc. But I don't think we've discussed a "complex install" flag and I think it could actually be quite useful both on the hub and in the viewer - we should capture this idea.

**Comment by junxini** [comment]
> not totally following here. _no_ plugins come pre-installed with the bundled app. And for those that are "uninstallable" can't we just not show them in the available list? (or, if you're trying to avoid someone from saying "why isn't this plugin showing up??" maybe show them grayed out with an icon on them?)

@tlambert03 thank you Talley for clarifying! My original description wasn't clear, it's been updated since. Hopefully it makes more sense now

**Comment by liaprins-czi** [comment]
> To summarize, plugins that can’t be installed will look like other plugins listed except for the following styling:
> - Add standard napari warning icon (with tool tip)
> - Install button is disabled
> - If possible, link out to troubleshooting plugins that can't be installed.

Overall this sounds / looks good to me! One thing that might also help to quickly communicate that it cannot be installed, and was mentioned earlier in the conversation, could be to grey out the plugin name text (or even change the background color subtly) so it looks less high-contrast with the dark background / looks disabled. I think that is pretty intuitive to most users to communicate that they cannot click on something or expect the standard behavior from it. It also would make it quicker to scan the list and see only the ones that could be installed. I still like the yellow triangle icon, in addition, but on its own it might not quite be enough to communicate quickly that the plugin cannot be installed (user might initially think it is just broken or didn't load in correctly or something else...).

**Comment by isabela-pf** [comment]
I posted some feedback on the related PR with mockups on the greyed-out overlay for uninstallable plugins. Just commenting here so anyone not following the PR is aware. [My comment on #4074](https://github.com/napari/napari/pull/4074#issuecomment-1058691987).
